### PR TITLE
Framed support classes

### PR DESCRIPTION
**Pull request by yschimke**
asan turned off as its failing when on, need to discuss.

I also added a couple of EXPECT_CALL to avoid noisy warnings. I know it can work without.

### CODE REVIEW

**Comment by stupaq-fb** [comment]
If ASAN build is failing then it's probably not a good idea to merge it. Do you have a paste with ASAN trace?

**Comment by yschimke** [comment]
It doesn't give much that I know how to use

```
==75025==ERROR: AddressSanitizer: heap-use-after-free on address 0x61400000ffe8 at pc 0x00010eb33c91 bp 0x7fff510e0450 sp 0x7fff510e0448
READ of size 8 at 0x61400000ffe8 thread T0
```

And was failing before I made any other changes.

**Comment by stupaq-fb** [comment]
Do you mean master was failing? It looked healthy...

**Comment by yschimke** [comment]
I think these are fixed now, looks like only leaks in the test code and related to calling a method on the heap allocated ref counted mocks after the Object Under Test destructor had already caused the refcount to go to 0.

File: reactivesocket-cpp/src/framed/FramedWriter.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <reactivesocket-cpp/src/streams/SmartPointers.h>
+#include "reactivesocket-cpp/src/Payload.h"
+#include "reactivesocket-cpp/src/ReactiveStreamsCompat.h"
+
+namespace reactivesocket {
+
+class FramedWriter : public reactivesocket::Subscriber<Payload>, public reactivesocket::Subscription {
```
Comment (stupaq-fb) [inline reactivesocket-cpp/src/framed/FramedWriter.h]: That looks long, did you run clang-format?

Comment (yschimke) [inline reactivesocket-cpp/src/framed/FramedWriter.h]: it didn't care

**Pull request merged**